### PR TITLE
build(ci): workflow improvements

### PR DESCRIPTION
**Pull request by sambacha**
## CI Workflow Improvements 

- adds Slither CI workflow
- provides basic hardhat configuration
- fixes pre-commit husky hook for running lint/formatting
- updates various nodejs dependencies 
- provides npm packaging for non-brownie usage (this is needed for Slither)
- small spelling corrections

## Formatting Issue

This specific section of the contract `BaseRouter.sol` was made by the prettier linter after complaining in my initial CI commit that it failed the `lint check` 

https://github.com/sambacha/yearn-vaults/commit/bc7bbb584af6246bbafe2c6cc696459586762b67#diff-25be7979238705af29ae97dc777ccbc048b0a1a56c4bf6c4021c6f4e7b8ac507R222-L229

offending CI process: https://github.com/sambacha/yearn-vaults/runs/3843829402?check_suite_focus=true#step:7:7

**wat do?**

## Slither Workflow

Where to find slither coverage results and how to observe reported slither issues. Note: probably mostly over eager reporting here.

### CODE REVIEW

**Comment by pandadefi** [comment]
I am not able to merge it because off the two missing checks @fubuloubu

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -93,6 +98,7 @@ jobs:
         run: pip install -r requirements-dev.txt
 
       - name: Run black
+        id: format:black
         run: black --check --include "(tests|scripts)" .
 
 # TODO: Add Slither Static Analyzer
```
Comment (pandadefi) [inline .github/workflows/lint.yaml:104]: Can be removed

**Comment by pandadefi** [comment]
I think it's missing instructions on how to run slither on local machines

**Comment by sambacha** [comment]
> The yarn.lock is still missing.

fixed in [`9983cde` (#473)](https://github.com/yearn/yearn-vaults/pull/473/commits/9983cdeea33e68c85c1fcdd12c60efc3cb6cc506)

**Comment by sambacha** [comment]
> I think it's missing instructions on how to run slither on local machines

I would think that is beyond the scope of this CI workflow, unless you want to support multiple dev environments explicitly? Let me know what operating systems you would like tutorials for and I can arrange that.

**Comment by sambacha** [comment]
Just to summarize @pandadefi the non-slither CI fixes pertain to nodejs, with specific fixes for solhint/prettier-plugin-solidity compatibility and some configuration file fixes

**Comment by sambacha** [comment]
I have created a separate workflow to illustrate some of slithers new functionality:

### Generate Data Dependency Tables

 


### ERC Conformance



### Summary of Static Analysis 



Additionally can generate SARIF output too 

@fubuloubu what sort of reporting would you prefer?

**Comment by sambacha** [comment]
also @pandadefi if you do not use `--exclude-dependencies --compile-force-framework hardhat` you will get slient failures as seen here: https://github.com/sambacha/yearn-vaults/runs/4132400886?check_suite_focus=true#step:8:54

**Comment by pandadefi** [comment]
Can you fix the commit lint and the linter?

**Comment by sambacha** [comment]
> Can you fix the commit lint and the linter?

fixed

**Comment by pandadefi** [comment]
> > Can you fix the commit lint and the linter?
> 
> fixed

Still two CI errors.

**Comment by sambacha** [comment]
> > > Can you fix the commit lint and the linter?
> > 
> > 
> > fixed
> 
> Still two CI errors.

How do you suggest I edit the commits? or should I create a fresh branch, squash commits + requested fixes and open a new PR?

**Comment by pandadefi** [comment]
You can `git reset HEAD~10 -i` to edit the commits, squash then all into one and fix the message.

**Pull request closed without merging**